### PR TITLE
MULE-10753: SDK: Inconsistent behavior when injecting a tlsContext in…

### PR DESCRIPTION
**Pull request by pablomusumeci**
… a pojo

### CODE REVIEW

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/ObjectTypeParameterParser.java [hunk 1]
```diff
@@ -85,13 +88,22 @@ protected void doParse(Builder definitionBuilder) throws ConfigurationException
     type.getFields().forEach(this::parseField);
   }
 
+  private String getFieldName(ObjectFieldType objectFieldType, String fieldName) {
+
+    return objectFieldType.getValue().getAnnotation(TypeIdAnnotation.class)
```
Comment (alegmarra) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/ObjectTypeParameterParser.java]: `InfrastructureTypeMapping.getNameMap().getOrDefault(ExtensionMetadataTypeUtils.getId(field.getValue()), fieldName)`
+ static imports

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/ObjectTypeParameterParser.java [hunk 2]
```diff
@@ -85,13 +88,22 @@ protected void doParse(Builder definitionBuilder) throws ConfigurationException
...
   private void parseField(ObjectFieldType objectField) {
     final MetadataType fieldType = objectField.getValue();
-    final String fieldName = objectField.getKey().getName().getLocalPart();
+    final String key = objectField.getKey().getName().getLocalPart();
+    final String fieldName = getFieldName(objectField, key);
     final boolean acceptsReferences = acceptsReferences(objectField);
     final Object defaultValue = getDefaultValue(fieldType).orElse(null);
     final ExpressionSupport expressionSupport = getExpressionSupport(objectField);
-    Optional<DslElementSyntax> fieldDsl = typeDsl.getChild(fieldName);
+    Optional<DslElementSyntax> fieldDsl = typeDsl.getChild(key);
```
Comment (pablomusumeci) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/ObjectTypeParameterParser.java]: I'm not really sure about doing that. An infrastructure parameter is kinda special, but it is still a parameter. You are going to have a different key name but the parsing is all the same. I know that we can shortcut it and go straight to the `parseObject(...)` thing, but if the parameter parsing changes in the future, we are still going to want those infrastructure parameters to be parsed in the same way as the others.

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/ObjectTypeParameterParser.java (hunk 1 above)
Comment (pablomusumeci) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/ObjectTypeParameterParser.java]: You will end up needing the .orElse() to deal with the case in which the objectField doesn't have the TypeIdAnnotation.

File: modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/ObjectTypeParameterParser.java [hunk 3]
```diff
@@ -85,6 +86,11 @@ protected void doParse(Builder definitionBuilder) throws ConfigurationException
     type.getFields().forEach(this::parseField);
   }
 
+
+  private Optional<String> getInfrastructureParameterName(MetadataType fieldType) {
+    return Optional.ofNullable(getNameMap().get(getId(fieldType)));
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/parameter/ObjectTypeParameterParser.java]: obtain the namaeMap once, and cache it into a field with a descriptive name

**Pull request merged**